Makefile

```diff
@@ -300,6 +300,9 @@ build/baserom/%.o: baserom/%
 build/asm/%.o: asm/%.s
```

Comment:
Couldn't we get rid of `asm/` related stuff entirely with this PR? For example this recipe is no longer needed, and `ASM_DIRS` could be reduced to `data/` and maybe renamed

Reply:
I left it in so it wouldn't need to be added back when `asm/` is put to use again for other versions, but I can remove it if you think it's better for now

Reply:
I considered that but I doubt `asm/` itself would be used for other versions. We would probably only ever need `asm/non_matchings` which isn't handled by the makefile. And either way it's easy enough to get it back from the git history if we do end up needing it